src/standalone/vector.rs

```diff
@@ -26,13 +29,51 @@ impl StandaloneVectorClient {
             rng: Arc::new(Mutex::new(ChaCha20Rng::from_entropy())),
         }
     }
+    pub(crate) async fn rotate_vector(
+        &self,
+        encrypted_vector: EncryptedVector,
+        metadata: &AlloyMetadata,
+        new_metadata: &AlloyMetadata,
+    ) -> Result<EncryptedVector, AlloyError> {
+        let (original_key_id, _) =
+            Self::decompose_encrypted_field_header(encrypted_vector.paired_icl_info.clone())?;
+
+        // check if we have a current secret for this path before doing significant work
+        // and that the current secret isn't the one this was encrypted with
+        let vector_secret = self
+            .config
+            .get(&encrypted_vector.secret_path)
+            .ok_or_else(|| {
+                AlloyError::InvalidConfiguration(format!(
+                    "Provided secret path `{}` does not exist in the vector configuration.",
+                    &encrypted_vector.secret_path.0
+                ))
+            })?;
+        let standalone_secret = vector_secret
+            .secret
+            .current_secret
+            .as_ref()
+            .ok_or_else(|| {
+                AlloyError::InvalidConfiguration(
+                    "No current secret exists in the vector configuration".to_string(),
+                )
+            })?;
 
-    /// The edek type for this client
+        if original_key_id.0 == standalone_secret.id {
```

Comment:
I think this also needs to check if the tenant IDs in the metadata are the same. If they aren't, we still need to do work